### ISSUE TITLE
Change

### ISSUE THREAD

**Issue by DominiqueMakowski** [Sussex-Psychology-Software-Team/DoggoNogo#7]
In a pilot that completed the game on the 7th, we lost some metadata fields (start, end, and stars/ends of L1):

![image](https://github.com/user-attachments/assets/be5c9fe2-884b-45cd-9fd2-7b153505fb41)

Is it related to some changes? @Max-Lovell

**Comment by Max-Lovell** [comment]
Just for a single participant?

**Comment by Max-Lovell** [comment]
nvm found the file - not sure, looking into it now. annoying we lost user agent too. Can I ask if anyone has been testing the code out? e.g. https://osf.io/s9u4d has been given NOT_BROWSER for a few fields - hopefully this was just one of your team testing things out in unity?

**Comment by DominiqueMakowski** [comment]
wdym by "testing the code out" Unless  @BenjaminTribe has been making some tests, I'd assume that the file is from a "true" pilot participant (?)

**Comment by DominiqueMakowski** [comment]
added you to the stufy's osf

**Comment by DominiqueMakowski** [comment]
can you tell us how to comment off sending data for testing purposes?

**Comment by Max-Lovell** [comment]
No worries that repo is just default so will always end up a bit messy. But yeah the missing start and end times is worrying. Data is sent on the script that controls the endScreen atm - this line https://github.com/Sussex-Psychology-Software-Team/DoggoNogo/blob/4b80ce21db2f6090cc4200920b048930ad670b6c/Assets/Scripts/End/EndScreen.cs#L62

**Comment by DominiqueMakowski** [comment]
Ok yes I think the problematic datafile is one from Ben's tests indeed as it doesn't have an associated experimental file (from jspsych), so I'll just delete it from the study OSF

**Comment by DominiqueMakowski** [comment]
Yeah no it's weird, the participant with the missing metadata for doggo also has a full experiment data (so it's indeed a true pilot)

**Comment by Max-Lovell** [comment]
Either way we need to know why it's missing for just one person. They have the url vars set fine (meaning it was an in-browser run, not in unity editor), but the fields relating to timing are all missing in metadata. The datetime for trials are in there, so it's not a problem with access to that function. The metadata datetime fields are also called all over the place, and all are missing, making it tricky to narrow down why. Hopefully we can replicate the bug somehow, but I'm a little stumped atm. There's a possibility it was a bug created during a small change at some point that it fixed now.

Also, there was a legacy bug where startL1 was being overwritten at the start of each level - these are incorrect in the pilot.'ve now called startL1 when participants press the spacebar to move on from the wooden sign instructions. I'm also going to suggest adding 'start' and 'end' date-time fields to each trial, and removing the current 'datetime' field (which tracks end time)? Consistent with metadata and potentially useful info, easy to add in.

**Comment by Max-Lovell** [comment]
Don't worry about this - figured out the issue.

End will be the datetime the trial is saved

**Comment by DominiqueMakowski** [comment]
> Don't worry about this - figured out the issue.

👍 

> End will be the datetime the trial is saved

which happens when?

**Comment by Max-Lovell** [comment]
Response given -> data processed -> data saved. During data saving an 'end' field is automatically filled out. A new trial will then begin, although not explicitly in an async manner or anything.

**Comment by Max-Lovell** [comment]
James has informed me a little refactor will be in order to properly sort this out and ensure things go smooth in future, be a day or two on this one. Are you currently collecting data?

**Comment by DominiqueMakowski** [comment]
nope still waiting for the ethics -_-

**Comment by DominiqueMakowski** [comment]
what do you plan on refactoring?

**Comment by DominiqueMakowski** [comment]
<sub>While you're at it, consider renaming the `Assets/Sausage/` folder for `Assets/Stimulus/` :p</sub>

**Comment by Max-Lovell** [comment]
How the data is handled into a Model-View-Controller design pattern. Will rename sausage to stim 👍

**Comment by DominiqueMakowski** [comment]
> how the data is handled into a Model-View-Controller design pattern

![](https://i.giphy.com/media/v1.Y2lkPTc5MGI3NjExNmpkeTU4bWFvZ2kwOTgwdnloazl5ZWk3YmZhcjNjNjM0a3Bnb25xdSZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/S5n7Wkhhw5A2IrfKER/giphy.gif)

**Comment by Max-Lovell** [comment]
You ask like I know

**Comment by Max-Lovell** [comment]
I kind of like having a sausage class too
